test/org/sagebionetworks/bridge/play/controllers/UserManagementControllerTest.java

```diff
@@ -87,6 +116,38 @@ public void before() throws Exception {
         when(request.headers()).thenReturn(map);
         doReturn(null).when(controller).getMetrics();
     }
+
+    @Test
+    public void signInForAdmin() throws Exception {
+        SignIn signIn = new SignIn.Builder().withStudy("originalStudy")
+                .withEmail(TestConstants.EMAIL).withPassword("password").build();
+        TestUtils.mockPlayContextWithJson(signIn);
+
+        when(authService.signIn(eq(study), any(CriteriaContext.class), signInCaptor.capture())).thenReturn(session);
+        
+        Result result = controller.signInForAdmin();
+        TestUtils.assertResult(result, 200);
+        
+        // This isn't in the session that is returned to the user, but verify it has been changed
+        assertEquals("originalStudy", session.getStudyIdentifier().getIdentifier());
+        assertEquals("api", signInCaptor.getValue().getStudyId());
+    }
+    
+    @Test
+    public void changeStudyForAdmin() throws Exception {
+        doReturn(session).when(controller).getAuthenticatedSession(Roles.ADMIN);
+        
+        SignIn signIn = new SignIn.Builder().withStudy("nextStudy").build();
+        TestUtils.mockPlayContextWithJson(signIn);
+        
+        Study nextStudy = Study.create();
+        nextStudy.setIdentifier("nextStudy");
+        when(studyService.getStudy("nextStudy")).thenReturn(nextStudy);
+        
+        controller.changeStudyForAdmin();
+        assertEquals("nextStudy", session.getStudyIdentifier().getIdentifier());
+        verify(cacheProvider).setUserSession(session);
```

Comment:
Do we also want to verify sessionUpdateService was called?

Reply:
It's a real object that has cache provider set as a mock; verifying cache provider and the session are updated, verifies the work that sessionUpdateService does (there's also a test of the method in that service in SessionUpdateServiceTest).

If this is too wonky, I can make sessionUpdateService a mock and just verify that it is being called instead.